swagger_server/test_local/test_ciupload_controller.py

```diff
@@ -0,0 +1,313 @@
+
+    def test_08_download(self):
+        """
+        Upload a document, encrypt, store, then recover, decrypt and make sure it's what
+        we started with ...
+        """
+        batch = str(uuid4())
+        with open('scripts/upload.txt', 'rb') as io:
+            fileobject = FileStorage(stream=io, filename='dummy.txt', name='myname')
+            #self.assertTrue(self.collection_instrument.upload(batch, fileobject, 'my_RU_code'))
+            code, msg = self.collection_instrument.upload(batch, fileobject, 'my_RU_code')
+            self.assertTrue(code == 200, msg)
+
+        code, msg = self.collection_instrument.instruments('{"ru_ref": "my_RU_code"}')
+        self.assertTrue(code == 200, msg)
+
+        code, msg = self.collection_instrument.download("my_RU_code")
+        with open('scripts/upload.txt', 'rb') as io:
+            orig = io.read()
+        self.assertEqual(msg, orig, 'Comparing processed with original')
+        code, msg = self.collection_instrument.download('does_not_exist')
+        self.assertTrue(code == 404, msg)
+        code, msg = self.collection_instrument.download(0)
+        self.assertTrue(code == 400, msg)
+
+    def test_09_misc(self):
+        """
+        Test misc (not required?) routines
+        """
+        batch = str(uuid4())
+        with open('scripts/upload.txt', 'rb') as io:
+            fileobject = FileStorage(stream=io, filename='dummy.txt', name='myname')
+            #self.assertTrue(self.collection_instrument.upload(batch, fileobject, 'my_RU_code'))
+            code, msg = self.collection_instrument.upload(batch, fileobject, 'my_RU_code')
+            self.assertTrue(code == 200, msg)
+
+        code, msg = self.collection_instrument.instruments('{"ru_ref": "my_RU_code"}')
+        print("Code>>>>", code)
+        self.assertTrue(code == 200, msg)
+        instrument = msg[0]
+        instrument_id = str(instrument['id'])
+
+        code, msg = self.collection_instrument.instrument(str(uuid4()))
+        print(">>>>>>>CODE IS", code)
+        self.assertTrue(code == 404, msg)
+
+        print("Instrument>", instrument_id)
+
+        code, msg = self.collection_instrument.instrument(instrument_id)
+        print("ICode>", code)
+        self.assertTrue(code == 200, 'Instrument fetch by id')
+
+        try:
+            code, msg = self.collection_instrument.instrument('abc')
+        except Exception:
+            pass
+        print("Code=", code)
+        self.assertTrue(code == 500, msg)
+
+        try:
+            code, msg = self.collection_instrument.instrument(0)
+        except Exception:
+            pass
+        print("Code=", code)
+        self.assertTrue(code == 400, 'Failed to lookup instrument with integer id')
+
+        try:
+            code, msg = self.collection_instrument.instrument()
+        except Exception:
+            pass
+        print("Code=", code)
```

Comment:
It will *always* return a code, the next line down should check / catch the correct status